src/full/Agda/Syntax/Parser/Parser.y

```diff
@@ -2403,8 +2404,15 @@ instance SetRange Attr where
 
 -- | Parse an attribute.
 toAttribute :: Expr -> Parser Attr
-toAttribute x = maybe failure (return . Attr (getRange x) y) $ exprToAttribute x
+toAttribute x = do
+  attr <- maybe failure (return . Attr r y) $ exprToAttribute x
+  case theAttr attr of
+    CohesionAttribute{} ->
+      modify' (\s -> s { parseCohesion = (y, r) : parseCohesion s })
+    _                   -> return ()
+  return attr
```

Comment:
I wonder if we shouldn't make note of _every_ `Attr` we come across? If we ever implement, e.g., #6246, this would be the place to do it.

Reply:
That does not need to be done now.